### DISCUSSION TITLE
[solved] Firestore-array-field into string via derivative-field

### DISCUSSION THREAD

**Discussion by webwurm** [AntlerVC/firetable#973] [Support / Q&A] [answered]
I'm sorry that I ask this question in a different way again - but it bothers me now for a few days and I can't figure it out. Maybe somebody has an idea...

### The situation
I have a field in firestore that is an array and stores the references to other documents. 
These documents have a field "Bezeichnung" (basically "title") and I want them in the current document put together as a string.

<img width="271" alt="Tags 01" src="https://user-images.githubusercontent.com/1053176/203508074-489a7dca-15c0-4120-a201-90d8186341a1.png">

<img width="373" alt="Tags 02" src="https://user-images.githubusercontent.com/1053176/203508099-0ea291c7-2316-429f-ba32-b95f79492fa0.png">

### My "solution"
I thought I do the following, since Rowy doesn't support an array in a field:
1. I get the array as JSON-field
2. In the derivative-field I cycle through the json and build the output-string

Sadly I don't even manage to get one item out of the JSON.
The json looks like this:
<img width="362" alt="Tags 03" src="https://user-images.githubusercontent.com/1053176/203508867-d1bdd247-9580-492e-8280-a3e990a4051c.png">

I tried so many things:
```
return row.tags.$[0]._key.path.segments[6];
return row.tags[0]._key.path.segments[6];
return row.tags.x[0]._key.path.segments[6];
```
...and many more versions; nothing works sadly.

### My questions
**So I have 2 questions:**
* Is it a good way what I am trying to do? Or is there a "smarter" way?
* How do I access the JSON, when the root-element is an unnamed array?

Thanks a lot!

**Comment by BeeBombshell** [comment, marked as answer]
Hey @webwurm, I tried recreating the issue you were facing. Firstly, I created a JSON field and set **Default as an array** to true. I replicated your `tags` data, as in the screenshot above, and stored it in a row.

<img width="314" alt="image" src="https://user-images.githubusercontent.com/53828745/203859716-77b1a05b-ed4b-4d8a-ac25-1f97344b67c9.png">

Then, I created another Derivative column and set its Output Field Type to Short Text. Following is the Derivative function I created for displaying a particular element of the `segments` array:

```js
const derivative:Derivative = async ({row,ref,db,storage,auth})=>{
    const data = row.tags[0]._key.path.segments[0]
    return data
  }
```
After deploying, the function seems to work perfectly and returns the correct value for the array element.
<img width="383" alt="image" src="https://user-images.githubusercontent.com/53828745/203860166-df64e282-2e77-4af6-b6c6-58cae185bfa7.png">

I wonder what seems to be the issue with your code. Could you share the complete JSON file so that I can help you figure this out?

**Reply by webwurm** [reply]
Thank you! I tried your solution and it still doesn't work. Here is the JSON, which is in a json-field with "array" checked.
[my.txt](https://github.com/rowyio/rowy/files/10089743/my.txt)

My derivative-field (shortText):
```
const derivative:Derivative = async ({row,ref,db,storage,auth})=>{
  const data = row.tags[0]._key.path.segments[0];
  return data
}
```

The error:
`Cannot read properties of undefined (reading 'path')`

**Reply by harinij** [reply]
Hi @webwurm if this is still an issue, perhaps you can join our upcoming [office hours](https://discord.gg/vZbFQzje?event=1059820599184195604) or book a [short 1:1 meeting](rowy.io/meet) so that we can go over this and get it resolved.

**Reply by webwurm** [reply]
I got it to work with putting the underline in the path somewhere else - although I don't know why it works, because in the JSON it is "_key.path..." and not "key._path":

`tagPath = "ChoiceChips/" + row.tags[i]._path.segments[1];`

But it works now, thank you!

**Reply by harinij** [reply]
Hi @webwurm FYI, we have a new array column type in case that is useful for your usecase.

Documentation: https://docs.rowy.io/field-types/array
Video: https://www.youtube.com/watch?v=-_zQjsIwPx8
